src/main/java/com/sapred/ordermanagerred/service/CurrencyConverterService.java

```diff
@@ -0,0 +1,45 @@
+package com.sapred.ordermanagerred.service;
+
+import com.sapred.ordermanagerred.dto.DailyGateKey;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import java.time.LocalDate;
+
+@Service
+public class CurrencyConverterService {
+
+    @Value("${exchangeRateApi}")
+    private String exchangeRateApi;
+    @Autowired
+    private RedisService redisService;
+
+    @SneakyThrows
+    public double convertCurrency(String fromCurrency, String toCurrency) {
+        String rate;
+        DailyGateKey dailyGateKey = new DailyGateKey(fromCurrency, toCurrency, LocalDate.now());
```

Comment:
I mean to save object as is in redis not convert to String
but also this syntax is good